report_scripts/list_pregnancy_women_receiving_in_clinic_events.sql

```diff
@@ -0,0 +1,68 @@
+-- List of visits by pregnant women to the clinic
+DROP PROCEDURE IF EXISTS list_pregnancy_women_receiving_in_clinic_events;
+DELIMITER $$
+CREATE PROCEDURE list_pregnancy_women_receiving_in_clinic_events(IN org_unit VARCHAR(11))
+BEGIN -- Vo6fHyvQhgJ TODO
+  DECLARE default_group_concat_max_len INTEGER DEFAULT 1024;
+  DECLARE max_group_concat_max_len INTEGER DEFAULT 4294967295;
+  DECLARE date_format VARCHAR(255) DEFAULT '%Y-%m-%d';
+  DECLARE program CHAR(11) DEFAULT 'mBMK6RRLKEZ';
+
+  SET SESSION group_concat_max_len = max_group_concat_max_len;
+
+  SELECT (SELECT CONCAT( "{\"events\": ", instances.entity_instance, "}")
+  FROM (SELECT CONCAT('[', instance.array, ']') as entity_instance
+    FROM (SELECT GROUP_CONCAT(entities_list.tracked_entity SEPARATOR ',') AS array
+      FROM (
+        SELECT DISTINCT JSON_OBJECT (
+          "program", program,
+          "programStage", "Vo6fHyvQhgJ",
+          "orgUnit", org_unit,
+          "eventDate", DATE_FORMAT(distinct_entity.start_date, date_format),
+          "status", "COMPLETED",
+          "storedBy", "admin",
+          "trackedEntityInstance", distinct_entity.program_patient_id,
+          "dataValues", JSON_ARRAY(
+            JSON_OBJECT(
+              "dataElement", "BruXV0FD2XS", -- No. de patient attribué par le site
+              "value", distinct_entity.st_id
+            ),
+            JSON_OBJECT(
+              "dataElement", "bzpXF1yVV74", -- No. dentité nationale
+              "value", distinct_entity.national_id
+            ),
+            JSON_OBJECT(
+              "dataElement", "vHkw3Habii4", -- Sexe
+              "value", distinct_entity.gender
+            ),
+            JSON_OBJECT(
+              "dataElement", "uSSFtn7oU2n", -- Age
+              "value", distinct_entity.age
+            ),
+            JSON_OBJECT(
+              "dataElement", "VT5fvNKFHr7", -- Date visite
+              "value", distinct_entity.start_date
+            )
+          )
+        ) AS tracked_entity
+        FROM (SELECT DISTINCT p.st_id, p.national_id, p.given_name, p.family_name,
+                p.gender, TIMESTAMPDIFF(YEAR, p.birthdate, NOW()) AS age, pp.start_date,
```

Comment:
Is this report the same situation as we had with the consultationByDay report? When you add the pp.start_date field to SELECT section we will get multiple events.

Reply:
You're right, we should find another way convert report like this.